finality-aleph/src/data_io/proposal.rs

```diff
@@ -29,6 +38,15 @@ pub struct UnvalidatedAlephProposal<B: BlockT> {
     pub number: NumberFor<B>,
 }
 
+/// Represents possible invalid states as described in [UnvalidatedAlephProposal].
+#[derive(Debug, Display, PartialEq, Error)]
+pub enum ValidationError {
+    BranchTooLarge,
```

Comment:
Just a suggestion: we could consider adding some numbers to the fields here to make it slightly more meaningful, e.g. the actual branch length and the limit here. Not necessary though, thinking out loud ;)

Reply:
Yeah, was thinking about it. Guess I'll do it after all.

Reply:
:heavy_check_mark: Added details to the errors